### PR TITLE
search delete get post manager

### PR DESCRIPTION
**Pull request by mansurissa**

### CODE REVIEW

File: controllers/employeeController.js [hunk 1]
```diff
@@ -108,3 +110,20 @@ export const employeeSuspend = async (req, res) => {
     });
   }
 };
+
+export const employeeSearch = async (req, res) => {
+  try {
+    const [keys] = Object.keys(req.body);
+    const [values] = Object.values(req.body);
+    const result = await Employees.find({
+      [keys]: { $regex: values },
```
Comment (RedJanvier) [inline controllers/employeeController.js:119]: can you please fix the issue! I was not able to search by phone number before merge.
An alternative you can try is: 
```
      - Find all employees with no condition
      - Filter the employees before sending the response
```

File: controllers/managersController.js [hunk 2]
```diff
@@ -1,16 +1,28 @@
...
-      manager,
+    await bcrypt.hash(password, 10, (err, hash) => {
+      if (err) {
+        console.log(err);
+        res.status(500).json({
+          message: 'auth failed',
+        });
+      } else {
+        console.log(hash);
+        const manager = Managers.create({
```
Comment (RedJanvier) [inline controllers/managersController.js:16]: please provide all managers details i.e: ​employee name, national id, phone number, date of birth, status​(active, inactive)​ and position​(manager)

File: controllers/managersController.js [hunk 3]
```diff
@@ -20,4 +32,58 @@ const managersSignup = async (req, res) => {
...
+    });
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({
+      message: 'failed to get all managers',
+    });
+  }
+};
+
+export const login = async (req, res) => {
```
Comment (RedJanvier) [inline controllers/managersController.js:50]: not getting any response on login

File: controllers/managersController.js (hunk 2 above)
Comment (RedJanvier) [reply controllers/managersController.js:16]: please check your response on manager signup! 
to fix it you can add an await on Manager.create()

File: models/employeesModel.js [hunk 4]
```diff
@@ -15,5 +15,10 @@ const employeesSchema = mongoose.Schema({
   suspended: { type: Boolean, default: false },
   phone: { type: Number, required: true, unique: true },
 });
-
+employeesSchema.index({
```
Comment (RedJanvier) [inline models/employeesModel.js:18]: Is this neccessary??

File: controllers/employeeController.js (hunk 1 above)
Comment (RedJanvier) [reply controllers/employeeController.js:119]: you can also provide a number of found results as an addition

File: models/employeesModel.js (hunk 4 above)
Comment (RedJanvier) [reply models/employeesModel.js:19]: remove it please!!!

File: controllers/employeeController.js (hunk 1 above)
Comment (mansurissa) [reply controllers/employeeController.js:120]: provided a number of found matches but searching by phone is not yet done

File: controllers/managersController.js (hunk 2 above)
Comment (mansurissa) [reply controllers/managersController.js]: thanks, i just corrected this

File: controllers/managersController.js (hunk 3 above)
Comment (mansurissa) [reply controllers/managersController.js:53]: response now is provided, thanks anyways

File: models/employeesModel.js (hunk 4 above)
Comment (mansurissa) [reply models/employeesModel.js:19]: found it unnecessary and removed it. thank

File: controllers/managersController.js (hunk 2 above)
Comment (RedJanvier) [reply controllers/managersController.js]: what about the phone number search ??

File: controllers/employeeController.js (hunk 1 above)
Comment (RedJanvier) [reply controllers/employeeController.js:120]: awesome job keep looking for another way to search for numbers. I will try to provide you with some help

Comment (mansurissa) [reply controllers/employeeController.js:120]: done but faked

**Pull request merged**